### PR TITLE
[fix]: [SRM-12188]: fix the previous analysis missing in the log analysis

### PR DESCRIPTION
**Pull request by AbhijithHarness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40258)

### CODE REVIEW

**Review by dhruv-harness — DISMISSED** [review]
nit: We can use CollectionUtils.emptyIfNull to do null check on collections.

> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40258#-:-NH0khgP6pMCqwszzEMa:b-nmz3q4)* status: :shipit: complete! ✓

File: 300-cv-nextgen/src/test/java/io/harness/cvng/analysis/resources/LogAnalysisResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    LearningEngineTask canaryLogAnalysisLearningEngineTask = builderFactory.canaryLogAnalysisLearningEngineTaskBuilder()
+                                                                 .uuid("")
+                                                                 .verificationTaskId(verificationTaskId)
+                                                                 .build();
+
+    learningEngineTaskId = learningEngineTaskService.createLearningEngineTask(canaryLogAnalysisLearningEngineTask);
+
+    String sampleDeploymentAnalysisDTOJson = getResource("analysis/sample-deployment-log-analysis-dto.json");
+    Response saveResponse = RESOURCES.client()
+                                .target("http://localhost:9998/log-analysis/deployment-save-analysis")
```
Comment (n00bitax) [inline 300-cv-nextgen/src/test/java/io/harness/cvng/analysis/resources/LogAnalysisResourceTest.java:90]: we should declare the resource URL at the top.`

**Review by deepak-harness — APPROVED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40258)* status: :shipit: complete! ✓

File: 300-cv-nextgen/src/test/java/io/harness/cvng/analysis/resources/LogAnalysisResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    hPersistence.save(verificationTask);
+  }
+
+  @Test
+  @Owner(developers = ABHIJITH)
+  @Category(UnitTests.class)
+  @SneakyThrows
+  public void testSaveDeploymentAnalysisAndGetPreviousDeploymentAnalysis() {
+    LearningEngineTask canaryLogAnalysisLearningEngineTask = builderFactory.canaryLogAnalysisLearningEngineTaskBuilder()
+                                                                 .uuid("")
```
Comment (n00bitax) [inline 300-cv-nextgen/src/test/java/io/harness/cvng/analysis/resources/LogAnalysisResourceTest.java:82]: as a good practice we should always be setting field like id , uuid

**Comment by AbhijithHarness** [comment]
trigger messagecheck

**Comment by AbhijithHarness** [comment]
trigger messagecheck

**Pull request merged**